### PR TITLE
Enhance processZipFile function with error handling and security improvements - Zip Slip

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Fixed the zip slip

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling and security for processing ZIP files.
	- Added validation to prevent path traversal vulnerabilities.

- **Bug Fixes**
	- Improved error messages for file operations and refined cleanup processes.

### CODE REVIEW

File: cmd/helpers/fileLoader.go [hunk 1]
```diff
@@ -79,24 +81,65 @@
...
+		outFile, err := os.Create(absExtractedPath)
 		if err != nil {
-			return nil, fmt.Errorf("failed to create file %s: %w", extractedFilePath, err)
+			return nil, fmt.Errorf("failed to create file %s: %w", absExtractedPath, err)
 		}
 		defer outFile.Close()
-		_, err = io.Copy(outFile, rc)
-		if err != nil {
-			return nil, fmt.Errorf("failed to copy file %s: %w", extractedFilePath, err)
+
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/helpers/fileLoader.go:140]: Ignore this

File: cmd/helpers/fileLoader.go [hunk 2]
```diff
@@ -79,24 +81,65 @@ func processZipFile(filePath string) ([]Data, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to create temp directory: %w", err)
 	}
-	defer os.RemoveAll(tempDir)
+	// Ensure the temp directory is removed in case of an error.
+	defer func() {
+		if err != nil {
+			os.RemoveAll(tempDir)
+		}
+	}()
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/helpers/fileLoader.go:89]: Ignore this

File: cmd/helpers/fileLoader.go [hunk 3]
```diff
@@ -79,24 +81,65 @@
...
+
+		absTempDir, err := filepath.Abs(tempDir)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get absolute path for temp directory: %w", err)
+		}
+
+		// Ensure that the extracted path is within the temp directory.
+		if !strings.HasPrefix(absExtractedPath, absTempDir+string(os.PathSeparator)) {
+			return nil, fmt.Errorf("invalid file path %s: outside of the extraction directory", f.Name)
+		}
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/helpers/fileLoader.go:115]: Ignore this

File: cmd/helpers/fileLoader.go [hunk 4]
```diff
@@ -79,24 +81,65 @@
...
+			if err := os.MkdirAll(filepath.Dir(absExtractedPath), os.ModePerm); err != nil {
+				return nil, fmt.Errorf("failed to create directory for file %s: %w", absExtractedPath, err)
+			}
+		}
+
 		rc, err := f.Open()
 		if err != nil {
 			return nil, fmt.Errorf("failed to open file %s in zip: %w", f.Name, err)
 		}
 		defer rc.Close()
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/helpers/fileLoader.go:134]: Ignore

**Review by neilnaveen — APPROVED** [review]

**Pull request merged**